server/src/main/java/org/candlepin/model/AsyncJobStatusCurator.java

```diff
@@ -86,4 +89,30 @@ public AsyncJobStatusCurator() {
 
         return this.getJobsInState(states);
     }
+
+    @Transactional
+    public AsyncJobStatus cancel(String jobId) {
+        this.cancelNoReturn(jobId);
```

Comment:
Given that we're actually returning the entity here, it probably makes sense to not hit the DB twice for this. Better off fetching, updating the status, persisting and then returning.

Reply:
Done.

Reply:
This is still doing a bit of unnecessary work, and introduces a bit of a race condition depending on the caching mode and innerworkings of Hibernate. It's fine for this specific PR, but will have to be completely reworked in the task that addresses the current DB vs messaging issue.